### ISSUE TITLE
After reloading a previous save, Bulletin Board is empty

### ISSUE THREAD

**Issue by lolo101** [pioneerspacesim/pioneer#5886]
### Observed behaviour

After landing at a station, I reloaded a save where I was in another system. Then I fly to the same station again and this time the Bulletin Board is empty
![image](https://github.com/user-attachments/assets/f80209a7-2850-436f-b7d7-ff18f3fc4e25)

### Expected behaviour

The Bulletin Board should show some missions, ads and so on

### Steps to reproduce

Load the attached savegame and open the station's BB

__My pioneer version (and OS):__

20240710 (e123a3f91) (Linux)

__My output.txt (required) and game save (optional, but recommended)__

[output.txt](https://github.com/user-attachments/files/16511417/output.txt)
[lolo1.gz](https://github.com/user-attachments/files/16511419/lolo1.gz)

**Comment by impaktor** [comment]
Strange / new one. 
I'd be surprised if it has anything to do with an issue as old as 899, since populating the BBS adverts was re-written just 1-2 years ago, and the UI has been completely re-written since then as well - twice I think, since that must be OldUI (-> NewUI -> PiGui).

**Comment by zonkmachine** [comment]
I've seen this before and I assumed it was related to https://github.com/pioneerspacesim/pioneer/issues/5305 which I was testing at the time . When I load the save in `lolo1.gz` and let it run for a while, the ship market is updated as it should but there are no new ads in the BBS ever so it's properly crashed.

Since the ship market is updated and `updateShipsOnSale(station)` and `updateAdverts(station)` are called at the same time we know that call is made.

I checked the Lua console and this logwarning is firing:
**"SpaceStation.lua: updateAdverts called for station that hasn't been visited"**

When I comment out all lines except `Event.Queue("onUpdateBB", station) ` in updateAdverts(), the BBS starts to repopulate itself.
https://github.com/pioneerspacesim/pioneer/blob/e123a3f91518ff4d0c84f90270803bf87dd2848c/data/libs/SpaceStation.lua#L901-L907

**Comment by zonkmachine** [comment]
Maybe `updateAdverts()` should look something like:

```lua
local function updateAdverts (station)
	if not SpaceStation.adverts[station] then
		logWarning("SpaceStation.lua: updateAdverts called for station that hasn't been visited")
	end
	Event.Queue("onUpdateBB", station)
end
```

**Comment by Web-eWorks** [comment]
If there isn't a `SpaceStation.adverts[station]` then the station's cached data needs to be re-created and new BBS adverts generated, yes. This is an edge-case scenario which suggests there is a deeper bug causing the station to be in an invalid / partially-valid state.

**Comment by zonkmachine** [comment]
About my fix above, From #5894, 

> This fix will cause the BBS to repopulate itself but only from those modules that impements onUpdateBB(). 

Right. If we can repopulate the BBS with `onUpdateBB`, can't we as well recreate it completely with `onCreateBB`?

```lua
 local function updateAdverts (station) 
 	if not SpaceStation.adverts[station] then 
 		logWarning("SpaceStation.lua: updateAdverts called for station that hasn't been visited") 
 		Event.Queue("onCreateBB", station) 
 	else 
 		Event.Queue("onUpdateBB", station) 
 	end 
 end
```

@Web-eWorks

**Comment by Web-eWorks** [comment]
Yes, `onCreateBB` is the correct function to use if there is no bulletin board data.

**Comment by zonkmachine** [comment]
> Yes, `onCreateBB` should be the correct function to use if there is no bulletin board data.

OK, fixed in: https://github.com/pioneerspacesim/pioneer/pull/5912
